### PR TITLE
:sparkles:  Add basic batteries

### PR DESCRIPTION
**Pull request by mjudeikis**
## Summary

Adds basic batteries to the control plane. This is not final state, but more to try and see what block are turnabble off and which ones needs some more upstream changes. This way we can experiment easier and get to proper decomposed control planes

## Related issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,29 @@ export KUBECONFIG=.gcp/admin.kubeconfig
...
+
+Batteries:
+- `leases` - a Kubernetes lease resources from `coordination.k8s.io`
+- `authentication` - Kubernetes native authentication using `authentication.k8s.io`
+- `authorization` - Kubernetes native authorization using `authorization.k8s.io`
+- `admission` - Kubernetes native admission using `admissionregistration.k8s.io`
+- `flowcontrol` - Kubernetes native flow control using `flowcontrol.apiserver.k8s.io`
+
+
+When starting server without any flags, in-memory storage will be used and batteries will be disabled by default.
```
Comment (sttts) [inline README.md:50]: what does in-memory storage mean? You mean you wire through no-op informers I assume?

File: server/cmd/server.go [hunk 2]
```diff
@@ -39,6 +38,8 @@ import (
 	cliflag "k8s.io/component-base/cli/flag"
 	"k8s.io/component-base/cli/globalflag"
 	"k8s.io/component-base/logs"
+
+	apiextensionapiserver "k8s.io/apiextensions-apiserver/pkg/apiserver"
```
Comment (sttts) [inline server/cmd/server.go:42]: strange order

File: README.md (hunk 1 above)
Comment (mjudeikis) [reply README.md:50]: in-memory etcd - our usual standalone development mode.

File: server/cmd/server.go (hunk 2 above)
Comment (mjudeikis) [reply server/cmd/server.go]: at some point need to wire linters into this repo :/  still very much basic stuff only

File: README.md (hunk 1 above)
Comment (sttts) [reply README.md]: ack

**Comment by mjudeikis** [comment]
Ready to merge so we can interate ontop @sttts

**Pull request merged**